src/modules/IEC61131-3/BitwiseOperators/genbitbase_fbt.cpp

```diff
@@ -73,8 +68,8 @@ bool CGenBitBase::createInterfaceSpec(const char *paConfigString, SFBInterfaceSp
   if (paInterfaceSpec.mNumDIs < CFunctionBlock::scmMaxInterfaceEvents) {
 
     //create the data inputs
-    mDataInputNames = new CStringDictionary::TStringId[paInterfaceSpec.mNumDIs];
-    mDataInputTypeIds = new CStringDictionary::TStringId[paInterfaceSpec.mNumDIs];
+    mDataInputNames.reset(new CStringDictionary::TStringId[paInterfaceSpec.mNumDIs]);
+    mDataInputTypeIds.reset(new CStringDictionary::TStringId[paInterfaceSpec.mNumDIs]);
```

Comment:
I would suggest to use [make_unique](https://en.cppreference.com/w/cpp/memory/unique_ptr/make_unique) instead to avoid the explicit `new` and also make it clear you are not replacing an existing pointer value:
```suggestion
    mDataInputNames = std::make_unique<CStringDictionary::TStringId[]>(paInterfaceSpec.mNumDIs);
    mDataInputTypeIds = std::make_unique<CStringDictionary::TStringId[]>(paInterfaceSpec.mNumDIs);
```